### PR TITLE
Add cmds/exp/tftp

### PR DESCRIPTION
**Pull request by ChriMarMe**
Add tftp CLI as command to execute.

Flags:
- [ ] `-4`/`-6` - not implemented due to pack.ag/tftp library resolves dynamically
- [ ] `-l` (literal) support for windows file pathes (handles file paths with `:` differently)
- [ ] `-R <Val:Val>` force originating port to be in range of Val:Val (no support in library)
- [x] `-c <cmd> <args...>`
- [x] `-m <mode>`
- [ ] `-v` (verbose) - still not sure what verbose does - need to investigate

Commands available in interactive mode:
- [x] help, h, ? - Prints help for interactive mode
- [x] q, quit - Quits the program and returns to shell
- [x] ascii - Sets transfer mode to netascii
- [x] binary - Sets transfer mode to octet
- [x] mode <ascii/binary> - Sets mode according to argument
- [x] get <args...> - Retrieves files from server (if server has been set previously)
- [x] put<args...> - Uploads files to server (if server has been set previously)
- [x] connect <host> [<port>] - sets hostname/ip and port to connect to
- [ ] literal - Switches literal mode on/off
- [x] rexmt <val> - Sets the per-packet retransmission limit to <val>
- [x] status - Prints the hostname/ip, mode and status of program and server related attributes
- [ ] trace - Switches packet tracing mode on/off
- [ ] verbose - Switches verbode mode on/off

### CODE REVIEW

**Comment by ChriMarMe** [comment]
Direct command execution with -c is still missing, hence Draft/Work in progresss

**Comment by 10000TB** [comment]
Great to see this. Please make sure coverage does not drop. Thank you

File: cmds/core/tftp/client.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+package main
+
+import (
+	"io"
+
+	"pack.ag/tftp"
+)
+
+type Response interface {
+	Read([]byte) (int, error)
```
Comment (10000TB) [inline cmds/core/tftp/client.go:14]: here and everywhere, public vars / structs / function, need a comment in form of `<> does abc ...` e.g. 

```
// Response ....
type Resobse interface {
   ..
}
```

File: cmds/core/tftp/client.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+type Client struct {
+	*tftp.Client
+}
+
+type RealResponse struct {
+	*tftp.Response
+}
+
+func (r *RealResponse) Read(b []byte) (int, error) {
+	return r.Response.Read(b)
```
Comment (10000TB) [inline cmds/core/tftp/client.go:52]: Here and everywhere, please look into addressing the coverage warning from `Codecov / codecov/patch`

**Review by 10000TB — COMMENTED** [review]
left somme comments

**Review by rminnich — COMMENTED** [review]
tftp is known to be a shit protocol, but if people want it, I guess we can put it in. But maybe it belongs in 
cmds/sucks/tftp
or 
cmds/exp/tftp
or 
cmds/really?/tftp

File: cmds/core/tftp/client.go (hunk 1 above)
Comment (ChriMarMe) [reply cmds/core/tftp/client.go]: done

**Comment by 10000TB** [comment]
discussed offline, let's move this to `cmd/exp/`. TY

**Review by 10000TB — APPROVED** [review]

**Pull request merged**